src/ml/neural_net/mps_layer_conv_padding.m

```diff
@@ -0,0 +1,57 @@
+#include <ml/neural_net/mps_layer_conv_padding.h>
+
+@implementation ConvolutionPadding
+
++ (BOOL) supportsSecureCoding {
+  return true;
+}
+
+- (id) initWithParams:(int)paddingWidth
+        paddingHeight:(int)paddingHeight
+          strideWidth:(int)strideWidth
+         strideHeight:(int)strideHeight {
+  @autoreleasepool {
+    self = [self init];
+    mPaddingWidth = paddingWidth;
+    mPaddingHeight = paddingHeight;
+    mStrideWidth = strideWidth;
+    mStrideHeight = strideHeight;
+    return self;
+  }
+}
+
+- (nullable instancetype)initWithCoder:(NSCoder *)coder {
+  @autoreleasepool {
+    self = [self init];
+    return self;
+  }
+}
+
+- (MPSNNPaddingMethod) paddingMethod {
+  return MPSNNPaddingMethodCustom;
+}
+
+- (MPSImageDescriptor *)destinationImageDescriptorForSourceImages:(NSArray<MPSImage *> *)sourceImages
+                                                     sourceStates:(NSArray<MPSState *> *)sourceStates
+                                                        forKernel:(MPSKernel *)kernel
+                                              suggestedDescriptor:(MPSImageDescriptor *)inDescriptor {
+  MPSCNNConvolution *layer = (MPSCNNConvolution *) kernel;
+  
+  MPSOffset pad;
+  pad.x = (int) ((layer.kernelWidth/2) - mPaddingWidth);
+  pad.y = (int) ((layer.kernelHeight/2) - mPaddingHeight);
+  pad.z = 0;
+  
+  layer.offset = pad;
```

Comment:
It is surprising to me that this function modifies the `kernel` input. I think that would be very confusing for potential future users of this code

Reply:
Should I copy the kernel and modify it instead?